brainglobe_stitch/image_mosaic.py

```diff
@@ -348,6 +355,48 @@ def read_big_stitcher_transforms(self) -> None:
             stitched_position = stitched_translations[tile.id]
             tile.position = stitched_position
 
+    def calculate_overlaps(self) -> None:
+        """
+        Calculate the overlaps between the tiles in the ImageMosaic.
+        """
+        self.overlaps = {}
+        z_size, y_size, x_size = self.tiles[0].data_pyramid[0].shape
+
+        for tile_i in self.tiles[:-1]:
+            position_i = tile_i.position
+            tile_i.neighbours = []
+            for tile_j in self.tiles[tile_i.id + 1 :]:
+                position_j = tile_j.position
+
+                # Check for overlap in the x and y dimensions
+                # and that the tiles do not have the same tile_id
+                # and that the tiles are from the same channel
+                if (
```

Comment:
Took me a while to parse these conditions 🤣

Reply:
They're a mouthful!